neurobooth_os/gui.py

```diff
@@ -614,11 +619,11 @@ def main():
         logger.critical(f"An uncaught exception occurred. Exiting: {repr(e)}")
         logger.critical(e, exc_info=sys.exc_info())
         logger.critical("Stopping GUI (error-state)")
-        raise
+        raise e
     finally:
+        print("logging shutdown starting")
```

Comment:
Are these leftover debug statements or intentionally left in?

Reply:
Yeah, I left them in because I still don't know why it hangs instead of shutting down. I suspect that the shutdown in GUI tells the other servers to shut down, which causes a hang because STM can't handle two messages at once.